lib/classes/EnvLoader.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const dotenv = require('dotenv');
+const fileExistsSync = require('../utils/fs/fileExistsSync');
+const path = require('path');
+const _ = require('lodash');
+
+class EnvLoader {
+  constructor(serverless) {
+    this.serverless = serverless;
+  }
+
+  async load() {
+    const serverlessConfigFile = this.serverless.pluginManager.serverlessConfigFile;
+    if (serverlessConfigFile == null) return;
+
+    const stage =
+      this.serverless.processedInput.options.stage ||
+      _.get(serverlessConfigFile, 'provider.stage', 'dev');
+
+    // TODO: Should it be based on provided `servicePath` ?
+    const defaultEnvFilePath = path.join(process.cwd(), '.env');
+    const stageEnvFilePath = path.join(process.cwd(), `.env.${stage}`);
+
+    const doesDefaultEnvFileExists = fileExistsSync(defaultEnvFilePath);
+    const doesStageEnvFileExists = fileExistsSync(stageEnvFilePath);
+
+    const useDotenv = serverlessConfigFile.useDotenv;
+    if (useDotenv === true) {
+      if (doesStageEnvFileExists) {
+        dotenv.config({ path: stageEnvFilePath });
+      } else if (doesDefaultEnvFileExists) {
+        dotenv.config({ path: defaultEnvFilePath });
+      }
```

Comment:
I like that approach, however, that suggests that you'd like the behavior to be a bit different than the one in `serverless/components` - please see my question from the above

> By following the similar approach as serverless/components, only one .env file can be loaded - would it make sense to load both .env and .env.stage if there're present in a way that .env.stage will be loaded later to override overlapping variables from .env?

So, do you think that we should load both env files (default one first, then stage) if they exist?

Reply:
> If I read correctly from dotenv implementation, they return object with error property if file was not found: motdotla/dotenv@7301ac9/lib/main.js#L108

Do you suggest using that to avoid explicit checks if the file exists? If yes, then, unfortunately, we still have to check for the existence of files in case the `useDotenv` is not set to `true`.

> This also reminds me, that it'll be good to signal any parsing errors to the users. I think it'll be good to crash then with meaningful error.

That's a good idea, however, I couldn't really come up with any actual possible parsing errors from `dotenv.config` that are not related to nonexistent file under provided path. I still think it's a good idea to check for any error. How would you like the framework to handle it? Crash or rather emit a warning to the user and continue processing as "normal"?

Reply:
>  If yes, then, unfortunately, we still have to check for the existence of files in case the useDotenv is not set to true

Yes, but it's temporary, it's only for deprecation reporting and will be removed with next major

>  How would you like the framework to handle it? Crash or rather emit a warning to the user and continue processing as "normal"?

I think we should crash. If _env_ setup is broken it's unlikely things will go well